scripts/lib/uaa_sync_admin_users.rb

```diff
@@ -0,0 +1,198 @@
+require 'securerandom'
+require 'time'
+require 'set'
+require 'uaa'
+
+require 'pp'
+
+# Utility class to create or delete admin users.
+#
+# Manages  users members from the groups listed in
+# UaaSyncAdminUsers::DEFAULT_ADMIN_GROUPS
+#
+class UaaSyncAdminUsers
+  attr_accessor :ua, :token_issuer, :admin_groups
+
+  DEFAULT_ADMIN_GROUPS = [
+    "cloud_controller.admin",
+    "uaa.admin",
+    "scim.read",
+    "scim.write",
+    "doppler.firehose"
+  ]
+
+  HOURS_TO_KEEP_TEST_USERS = 2
+
+  # Initialise the class with to the given UAA target with the given
+  # credentials.
+  #
+  # Params:
+  # - target: UAA target
+  # - admin_client: Admin client name for UAA with permissions to manage users
+  # - admin_password: Admin client password
+  # - options: Hash of options. Examples and defaults
+  #   - skip_ssl_validation: default=false
+  #   - extra_admin_groups: default=[] Additional admin groups apart of DEFAULT_ADMIN_GROUPS
+  #   - log_level: default: :warn. Options: :debug, :trace, :warn
+  def initialize(target, admin_client, admin_password, options=nil)
+    @target = target
+    @admin_client = admin_client
+    @admin_password = admin_password
+    @options = options
+
+    self.admin_groups = DEFAULT_ADMIN_GROUPS + (options[:extra_admin_groups] || [])
+
+    self.token_issuer = CF::UAA::TokenIssuer.new(@target, @admin_client, @admin_password, @options)
+    self.token_issuer.logger = self.get_logger()
+  end
+
+  # Returns the logger for this object.
+  # Set by options[:log_level] = {:debug, :trace, :warn} or $UAA_LOG_LEVEL
+  def get_logger()
+    if @logger.nil?
+      if ENV['UAA_LOG_LEVEL']
+        log_level = ENV['UAA_LOG_LEVEL'].strip.downcase.to_sym
+      else
+        log_level = @options[:log_level] || :info
+      end
+      @logger = Logger.new($stdout)
+      @logger.level = Logger::Severity.const_get(log_level.to_s.upcase)
+    end
+    @logger
+  end
+
+  def token=(token)
```

Comment:
token is an instance variable. `self.token =` can be replaced by `@token =` to access it internally instead of via the setter.
The tests pass with this change and without the setter method.